### PR TITLE
Planner 1580 add frontend service

### PR DESCRIPTION
**Pull request by MusaTalluzi**
Add front end with the ability to:
- Load submitted problems and view best solutions
- Add new problems (tenants)
- Switch between different tenants and view their best solutions

To run the ui:
- run spring boot app from the root directory:
-- `./mvnw spring-boot:run`
- run frontend app in a different terminal:
-- `cd spring-boot-task-assigning-frontend`
-- `npm start`

### CODE REVIEW

**Comment by rsynek** [comment]
Sorry for holding the review off; I might be able to get to it tomorrow or early next week.

File: spring-boot-task-assigning-frontend/src/components/ScheduleComponent.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import Scheduler, {
+  SchedulerData, ViewTypes, DATETIME_FORMAT, DATE_FORMAT,
+} from 'react-big-scheduler';
+import 'react-big-scheduler/lib/css/style.css';
+import moment from 'moment';
+
+import constants from '../shared/constants';
+
+// idToColor from "Joe Freeman" answer: https://stackoverflow.com/questions/3426404/create-a-hexadecimal-colour-based-on-a-string-with-javascript
+const idToColor = (str) => {
```
[the remark this answers is not in the document]
Comment (MusaTalluzi) [reply spring-boot-task-assigning-frontend/src/components/ScheduleComponent.js]: What's the motivation of moving it? I would've done so if it's to be used in another .js file or it's too long, but in this case it's a short function that's only relevant to `ScheduleComponent`, unless you have a stronger argument for moving it.

File: spring-boot-task-assigning-frontend/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+## OptaTask UI:
+In order to test the functionality, first have a KIE server running on http://localhost:8080 as explained here: https://docs.optaplanner.org/7.12.0.Final/optaplanner-wb-es-docs/html_single/#_planner.quickstart
```
[the remark this answers is not in the document]
Comment (MusaTalluzi) [reply spring-boot-task-assigning-frontend/README.md]: Actually this instruction is relevant to `kie-server-task-assignment` demo, I should remove it altogether.

File: spring-boot-task-assigning-frontend/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+
+## Available Scripts
+
+In the project directory, you can run:
+
+### `npm start`
```
[the remark this answers is not in the document]
Comment (MusaTalluzi) [reply spring-boot-task-assigning-frontend/README.md]: This readme is auto generated by create-react-app, I will put the instructions for unfamiliar useres in the parent directory README,

**Comment by MusaTalluzi** [comment]
@rsynek thanks for the review! There are no plans currently for an optaweb-task-assignment, and the UI is just for testing and experimental purposes for using SolverManager, that's why there are no tests.

File: spring-boot-task-assigning-frontend/src/components/ScheduleComponent.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rsynek) [reply spring-boot-task-assigning-frontend/src/components/ScheduleComponent.js]: Just not to spoil the logic of the component with this special detail. As it's not reused from anywhere else, it's not important.

**Pull request merged**